File: Python/Domotica/LampadinaSmart.py
from DispositivoSmart import DispositivoSmart
class LampadinaSmart(DispositivoSmart):
    def __init__(self, stato=False, consumo=0):
        self.stato = stato
        self.consumo = consumo
    def accendi(self):
        if not self.stato:
            self.stato = True
    def spegni(self):
        if self.stato:
            self.stato = False
    def get_consumo(self):
        return self.consumo if self.stato else 0

File: Java/domoticaJava/src/LampadinaSmart.java
public class LampadinaSmart implements DispositivoSmart {
    private boolean stato;
    private double consumo;

    public LampadinaSmart(double consumo) {
        this.stato = false;
        this.consumo = consumo;
    }
    @Override
    public void accendi() {
        stato = true;
        System.out.println("Lampadina accesa.");
    }
    @Override
    public void spegni() {
        stato = false;
        System.out.println("Lampadina spenta.");
    }
    @Override
    public double get_consumo() {
        return stato ? consumo : 0.0;
    }
}
